packages/@azure/servicebus/data-plane/test/testUtils.ts

```diff
@@ -39,104 +42,305 @@ export enum ClientType {
+  const env = await getEnvVars();
+  await msRestNodeAuth
+    .loginWithServicePrincipalSecret(env.clientId, env.secret, env.tenantId)
+    .then(async (creds) => {
+      const client = await new ServiceBusManagementClient(creds, env.subscriptionId);
+      await client.queues.deleteMethod(
+        env.resourceGroup,
+        env.servicebusNamespace,
+        queueName,
+        function(error: any): void {
+          if (error) throw error.message;
+        }
       );
-    case ClientType.PartitionedTopicWithSessions:
-      return namespace.createTopicClient(
-        process.env.TOPIC_NAME_SESSION || "partitioned-topic-sessions"
+      await client.queues.createOrUpdate(
+        env.resourceGroup,
+        env.servicebusNamespace,
+        queueName,
+        parameters,
+        function(error: any): void {
+          if (error) throw error.message;
+        }
       );
-    case ClientType.UnpartitionedQueueWithSessions:
-      return namespace.createQueueClient(
-        process.env.QUEUE_NAME_NO_PARTITION_SESSION || "unpartitioned-queue-sessions"
+    })
+    .catch((err) => {
+      console.log(err.message);
+    });
+}
+
+async function recreateTopic(topicName: string, parameters: SBTopic): Promise<void> {
+  const env = await getEnvVars();
+  await msRestNodeAuth
+    .loginWithServicePrincipalSecret(env.clientId, env.secret, env.tenantId)
+    .then(async (creds) => {
+      const client = await new ServiceBusManagementClient(creds, env.subscriptionId);
+      await client.topics.deleteMethod(
+        env.resourceGroup,
+        env.servicebusNamespace,
+        topicName,
+        function(error: any): void {
+          if (error) throw error.message;
+        }
       );
-    case ClientType.UnpartitionedTopicWithSessions:
-      return namespace.createTopicClient(
-        process.env.TOPIC_NAME_NO_PARTITION_SESSION || "unpartitioned-topic-sessions"
+      await client.topics.createOrUpdate(
+        env.resourceGroup,
+        env.servicebusNamespace,
+        topicName,
+        parameters,
+        function(error: any): void {
+          if (error) throw error.message;
+        }
       );
-    case ClientType.TopicFilterTestTopic:
-      return namespace.createTopicClient(process.env.TOPIC_FILTER_NAME || "topic-filter");
-    default:
-      break;
-  }
-
-  throw new Error("Cannot create sender client for give client type");
+    })
+    .catch((err) => {
+      console.log(err.message);
+    });
 }
 
-export function getReceiverClient(
+async function recreateSubscription(
+  topicName: string,
+  subscriptionName: string,
+  parameters: SBSubscription
+): Promise<void> {
+  const env = await getEnvVars();
+  await msRestNodeAuth
+    .loginWithServicePrincipalSecret(env.clientId, env.secret, env.tenantId)
+    .then(async (creds) => {
+      const client = await new ServiceBusManagementClient(creds, env.subscriptionId);
+      await client.subscriptions.createOrUpdate(
```

Comment:
We are not deleting the subscription here because before calling `recreteSubscription` we would have called `recreateTopic` which would have deleted the topic and so all subscriptions along with it?

If yes, then let's add a comment here for the same